docs/ruleset_functions/compare_swh_dist_systems_and_components.md

```diff
@@ -0,0 +1,54 @@
+## compare_swh_dist_systems_and_components
+
+Description: This function compares all sub-components of a given SWH Distribution System ID between two models.  The function also accepts   
+
+Inputs:
+- **RMD1**
+- **RMD2**
+- **compare_context_str**
+- **swh_distribution_id**
+- **errors**
```

Comment:
What does this errors do? Suggest explaining that here

Reply:
errors is a list that is one of the required inputs for compare_context_str.  Passing it through allows any errors reported by compare_context_str to be accessible by the rule that called this function

Reply:
Can return error list instead of boolean?
An empty list means full match and with message means there are diffs.

Reply:
@weilixu - this is an excellent idea.  I updated the function per your suggestion.  To confirm - does compare_context_str always append an error to the errors list when the result is false?

Reply:
> @weilixu - this is an excellent idea. I updated the function per your suggestion. To confirm - does compare_context_str always append an error to the errors list when the result is false?

That is correct.